pycbc/workflow/segment.py

```diff
@@ -1389,3 +1390,58 @@ def file_needs_generating(file_path, cp, tags=None):
             err_msg += '"always", "if_not_present", "error_on_duplicate", '
             err_msg += '"never". Got %s.' %(generate_segment_files,)
             raise ValueError(err_msg)
+
+def get_segments_file(workflow, name, option_name, out_dir):
+    """Get cumulative segments from option name syntax for each ifo.
+
+    Use syntax of configparser string to define the resulting segment_file
+    e.x. option_name = +up_flag1,+up_flag2,+up_flag3,-down_flag1,-down_flag2
+    Each ifo may have a different string and is stored separately in the file.
+
+    Parameters
+    ----------
+    workflow: pycbc.workflow.Workflow
+    name: string
+        Name of the segment list being created
+    option_name: str
+        Name of option in the associated config parser to get the flag list
+
+    returns
+    --------
+    seg_file: pycbc.workflow.SegFile
+        SegFile intance that points to the segment xml file on disk.
+    """
+    from pycbc.events.status import query_combined_flags
+    from pycbc.workflow import SegFile
+    make_analysis_dir(out_dir)
+    cp = workflow.cp
+    start = workflow.analysis_time[0]
+    end = workflow.analysis_time[1]
+
+    # Check for veto definer file
+    veto_definer = None
+    if cp.has_option("workflow-segments", "segments-veto-definer-url"):
+         veto_definer = save_veto_definer(workflow.cp, out_dir, [])
+
+    # Check for provided server
+    server = "segments.ligo.org"
+    if cp.has_option("workflow-segments", "segments-database-url"):
+        server = cp.get_opt("workflow-segments", "segments-database-url")
+    
+    segments = {}
+    for ifo in workflow.ifos:
+        flag_str = cp.get_opt_tags("workflow-segments", option_name, [ifo])
```

Comment:
So if H1 and L1 (for e.g.) are using different definitions for science (V1 certainly will for internal usage) we would need to have `[workflow-segments-h1]`, `[workflow-segments-l1]` and `[workflow-segments-v1]`? As this is a top-level `workflow` section, there's no reason to not allow the current `segments-science-h1` syntax, but either way it should be clear (e.g. make the example assume that H1/L1/V1 definitions are different even if they are not).

Reply:
.... I also deliberately kept the functionality that you can use H1 flags (and data) and tell the workflow it is V1 (for example). I think you've removed that ... It might be useful.

Reply:
(maybe I'm saying keep the old syntax and enforce a full segment string e.g. `H1:SCIENCE:1`)

Reply:
I think redundancy should be avoided where possible, but I'm not against allowing an easy override. I propose to allow the following to make this self-consistent.

H1:SCIENCE:1 will just use this flag for whatever detector you had said. 
SCIENCE:1 standard form, will use the ifo that is was requested a flag for